core/drivers/ffa_console.c

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION
+ */
+
+#include <compiler.h>
+#include <console.h>
+#include <drivers/ffa_console.h>
+#include <drivers/serial.h>
+#include <kernel/thread_arch.h>
+
+#define FFA_CONSOLE_LOG_32		(0x8400008A)
+
+static void ffa_console_putc(struct serial_chip *chip __unused, int ch)
+{
+	thread_hvc(FFA_CONSOLE_LOG_32, 1, ch, 0);
```

Comment:
The ABI supports printing more than one character at a time.
Can `thread_smccc()` be used instead to support an SPMC at EL3?